### PR TITLE
Add imageRegistry and imagePath support in Helm Chart

### PR DESCRIPTION
**Pull request by dthomson-sysdig**
## Description

Add in support for third-party (private) image registry and image path in the helm chart.

Testing has been done manually on a private k8s cluster with a private ECR.

Docs probably want to be updated for Helm installation so that users are aware of the feature flags.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by caseydavenport** [comment]
https://github.com/projectcalico/calico/pull/7395/files#diff-1ef25acab082b9c39db773ccde81f063b32519075a43897fddc882256bda701dR15

The helm chart values already allow you to put an arbitrary json spec in for the installation in order to configure these two fields, so I don't think we should have explicit call outs for them.

The design of this chart is that the `values.installation` field is free-form, and is directly passed as the spec to the Installation that's created on the cluster, so that we don't need to constantly add new fields to the chart as the API evolves.

In other words, the following yaml should already work:

```yaml
installation:
  imagePath: ""
  registry: ""
```

File: charts/tigera-operator/templates/crs/custom-resources.yaml [hunk 1]
```diff
@@ -16,6 +16,13 @@ spec:
 
 {{ end }}
 
+{{- if .Values.imageRegistry }}
```
Comment (caseydavenport) [inline charts/tigera-operator/templates/crs/custom-resources.yaml:19]: Per my other comment, I don't think we should add these fields. But as an aside, if we were to add them they would need to be above the `{{ end }}` clause on line 17. Otherwise, setting `values.installation.enabled` to false will result in the generation of an invalid custom resources file.

**Comment by dthomson-sysdig** [comment]
> https://github.com/projectcalico/calico/pull/7395/files#diff-1ef25acab082b9c39db773ccde81f063b32519075a43897fddc882256bda701dR15
> 
> The helm chart values already allow you to put an arbitrary json spec in for the installation in order to configure these two fields, so I don't think we should have explicit call outs for them.
> 
> The design of this chart is that the `values.installation` field is free-form, and is directly passed as the spec to the Installation that's created on the cluster, so that we don't need to constantly add new fields to the chart as the API evolves.
> 
> In other words, the following yaml should already work:
> 
> ```yaml
> installation:
>   imagePath: ""
>   registry: ""
> ```

ah sure, that's fair, @caseydavenport

File: charts/tigera-operator/templates/crs/custom-resources.yaml (hunk 1 above)
Comment (dthomson-sysdig) [reply charts/tigera-operator/templates/crs/custom-resources.yaml:19]: Right, makes sense.

**Comment by caseydavenport** [comment]
@dthomson-sysdig I'm going to close this one for now, but appreciate you putting in the time to submit this! Please let us know if there's some reason what I suggested above is insufficient and we can discuss more.

**Comment by lwr20** [comment]
If the above wasn't obvious, perhaps we need a PR in the docs to make it more obvious for the next person?

**Comment by dthomson-sysdig** [comment]
> If the above wasn't obvious, perhaps we need a PR in the docs to make it more obvious for the next person?

Hi @lwr20 , yes, I didn't catch that `installation` is free-form.

A mention for private registry/path on [https://docs.tigera.io/calico/3.25/getting-started/kubernetes/helm](https://docs.tigera.io/calico/3.25/getting-started/kubernetes/helm) would go a long way I reckon.

**Comment by lwr20** [comment]
@ctauchen FYI
